src/sources/static_metrics.rs

```diff
@@ -0,0 +1,312 @@
+use std::collections::BTreeMap;
+use std::num::NonZero;
+use std::time::Duration;
+
+use chrono::Utc;
+use futures::StreamExt;
+use serde_with::serde_as;
+use tokio::time;
+use tokio_stream::wrappers::IntervalStream;
+use vector_lib::configurable::configurable_component;
+use vector_lib::internal_event::{CountByteSize, InternalEventHandle as _};
+use vector_lib::{config::LogNamespace, ByteSizeOf, EstimatedJsonEncodedSizeOf};
+
+use crate::{
+    config::{SourceConfig, SourceContext, SourceOutput},
+    event::{
+        metric::{MetricData, MetricName, MetricSeries, MetricTime, MetricValue},
+        EventMetadata, Metric, MetricKind,
+    },
+    internal_events::{EventsReceived, StaticMetricsBytesReceived, StreamClosedError},
+    shutdown::ShutdownSignal,
+    SourceSender,
+};
+
+/// Configuration for the `static_metrics` source.
+#[serde_as]
+#[configurable_component(source(
+    "static_metrics",
+    "Produce static metrics defined in configuration."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct StaticMetricsConfig {
+    /// The interval between metric emitting, in seconds.
+    #[serde_as(as = "serde_with::DurationSecondsWithFrac<f64>")]
+    #[serde(default = "default_interval")]
+    #[configurable(metadata(docs::human_name = "Emitting interval"))]
+    pub interval_secs: Duration,
+
+    /// Overrides the default namespace for the metrics emitted by the source.
+    #[serde(default = "default_namespace")]
+    pub namespace: String,
+
+    #[configurable(derived)]
+    #[serde(default = "default_metrics")]
```

Comment:
You can use this 
```suggestion
    #[serde(default)]
```
if you specify only `default`, serde will use `std::default::Default` https://serde.rs/attr-default.html

Reply:
Thanks! Not sure why I made that separate function :thinking: